spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -131,9 +166,24 @@ protected DeleteCounter counter() {
 
   public boolean next() throws IOException {
     try {
+      nextMethodStartTime = System.currentTimeMillis();
+      LOG.info("at next method");
+      try {
+        completableFutureList = prefetchS3Files();
```

Comment:
why not directly call this when BaseReader is constructed?

Reply:
We should call it as early as possible, so maybe not even at the BaseReader constructor, maybe some place in the caller, basically as soon as when a SparkInputPartition is fed to the worker.

Reply:
https://github.com/apache/iceberg/blob/5584426b7de05b81624b0ff8d91f8891c69faaa8/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkColumnarReaderFactory.java#L43-L58 - probably inside here?